maindataclasses.rb

```diff
@@ -26,24 +37,26 @@
 puts array2.length
 
 # task 5
+puts "task5"
 array3 = [1,2,3,4,5,6,7,8,9]
 array4 = []
 array5 = []
 a = 1
 idx = array3.index(array3[-1])
 while a <= idx
-    if array3.index(array3[a]).odd?
-      array4 << array3[a]
-    elsif array3.index(array3[a]).even?
-      array5 << array3[a]
-    end
-    a += 1
+  if array3.index(array3[a]).odd?
+    array4 << array3[a]
+  elsif array3.index(array3[a]).even?
+    array5 << array3[a]
+  end
+  a += 1
 end
 puts array4
 puts array5
 
 
 # task 6
+puts "task6"
```

Comment:
you should to find the last element and puts it out
but you puts all elements where expected conditions been met